src/Storages/MergeTree/GinIndexStore.cpp

```diff
@@ -265,6 +229,20 @@ void GinIndexStore::finalize()
     if (!current_postings.empty())
         writeSegment();
 
+    // If next available segment id has value, then the value has to be saved to disk again with the updated segment id
+    if (next_available_segment_id.has_value())
```

Comment:
It would be nice to move l. 234-243 into its own function, analogous to `writeSegment()`, l. 230.